packages/core/src/common/messaging/proxy-factory.ts

```diff
@@ -240,7 +240,7 @@ export class JsonRpcProxyFactory<T extends object> implements ProxyHandler<T> {
                     try {
                         if (isNotify) {
                             connection.sendNotification(method, ...args);
-                            resolve();
+                            resolve(undefined);
```

Comment:
Rather change line 239 to be `new Promise<void>(...`

Reply:
No, the promise can yield a result if it's not a notification.

Reply:
OK I had not seen the code around it! Though is this required by TypeScript? It's often fine to just write `resolve()` when nothing is returned, and `resolve(undefined)` is usually a smell that the promise constructor is not aptly typed.